src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -155,7 +155,7 @@ BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input,
                 if (input.Type == type)
                 {
                     return _factory.Sequence(
-                        _factory.AssignmentExpression(_factory.Local(target), input),
+                        _factory.AssignmentExpression(target, input),
```

Comment:
Where the expectations are verified/enforced. Where they are documented. 

---
In reply to: [77397840](https://github.com/dotnet/roslyn/pull/13535#discussion_r77397840) [](ancestors = 77397840,77395353)